lib/proxy_api/external_ipam.rb

```diff
@@ -63,7 +63,7 @@ def initialize(args)
     #     {"error": "Unable to connect to External IPAM server"}
     def next_ip(subnet, mac, group = "")
       raise "subnet cannot be nil" if subnet.nil?
-      response = parse get("/subnet/#{subnet}/next_ip?mac=#{mac}&group=#{URI.escape(group.to_s)}")
+      response = parse get("/subnet/#{subnet}/next_ip?mac=#{mac}&group=#{Addressable::URI.encode(group.to_s)}")
```

Comment:
We don't depend on Addressable. At least not directly. This means there's no guarantee this gem is available.

I wonder if using `URI.encode_www_form` is correct. I think it would be something like this:
```suggestion
      response = parse get("/subnet/#{subnet}/next_ip?#{URI.encode_www_form(mac: mac, group: group.to_s)}")
```

Though I'm not 100% sure it is correct.

Reply:
Using URI.escape is the correct way of doing this now.

Reply:
I tried both of them and the output is different from what is expected.

Reply:
Could you share the examples you tried?

Reply:
I wasn't sure how to use the URI.encode_www_form because I have a different path at each time. 

This is what I checked: 
![image](https://user-images.githubusercontent.com/57755873/140930284-c11a983b-d688-41a3-a460-f27b88a64e3e.png)

And these are the errors:
 
* when using CGI.escape: 
![image](https://user-images.githubusercontent.com/57755873/140930178-750a10fb-466c-4c85-aa32-d5806e5c5831.png)

* when using  URI.encode_www_form_component:
![image](https://user-images.githubusercontent.com/57755873/140930723-0d66acfb-c081-4dab-904f-83e375fd5917.png)

Reply:
I wasn't sure how to use the URI.encode_www_form because I have a different path at each time.

This is what I checked:
![image](https://user-images.githubusercontent.com/57755873/140931634-b83def5a-57c6-45cd-ac7a-a6fe91451cc9.png)
 
* when using CGI.escape:
![image](https://user-images.githubusercontent.com/57755873/140930178-750a10fb-466c-4c85-aa32-d5806e5c5831.png)

* when using  URI.encode_www_form_component:
![image](https://user-images.githubusercontent.com/57755873/140930723-0d66acfb-c081-4dab-904f-83e375fd5917.png)

Reply:
For future reference: please copy-paste it rather than a screenshot. Consuming plain text is much easier than screenshots.

For your paths you should only encode the components, not the entire path.

Reply:
Katello had the same issue and they fixed by switching to `URI.encode_www_form` as @ekohl suggested, see [Katello - Fixes #32995 - URI.escape is obsolete warning](https://github.com/Katello/katello/pull/9477/commits/de337537b505f9a4447d230eb43205034223cf94#diff-6ff2fabe1c0b1475dcf25fedcb7546b9bc0336efc40e3c6443a2b2b43d0bd576R154)